reasoner_validator/biolink/__init__.py

```diff
@@ -317,14 +300,99 @@ def validate_edge(self, edge: Dict):
 
         if self.graph_type is TrapiGraphType.Knowledge_Graph:
             if not attributes:
-                # TODO: not quite sure whether and how to fully validate the 'attributes' of an edge
                 # For now, we simply assume that *all* edges must have *some* attributes
                 # (at least, provenance related, but we don't explicitly test for them)
                 self.report_error(f"Edge '{edge_id}' has missing or empty attributes?")
+            else:
+                # TODO: attempt some deeper attribute validation here
+                for attribute in attributes:
+                    attribute_type_id: str = attribute['attribute_type_id']
```

Comment:
is attribute.get('attribute_type_id', None) necessary here or can we assume this key exists due to prior validation?

Reply:
yes, good idea